### DISCUSSION TITLE
ability to split yaml documents into specific output files

### DISCUSSION THREAD

**Discussion by m0un10** [k14s/ytt#471] [Ideas]
I understand that `ytt` was designed very much with the idea of piping to k8s in mind. That said, I am really finding it useful for generating any yaml documents that are written to the file system. I would really love it if there was a way to output to specific files. The `output-files` does a good job of creating separate files that match the input files. However, it won't create separate files for each document (split by `---`). I was considering to build a tool that takes piped yaml and looks for `---` and then uses metadata to determine where the files should be written to. Is this something you would ever want to be able to do in `ytt` or would I be better to look at creating a new tool that works alongside `ytt` for this specific use case?

**Comment by m0un10** [comment]
The benefit of something like this is it would allow me to model all the data in YTT `data/values` even if there is a for loop to create multiple documents. Without something like this, it just means that the primary data source shifts outside YTT and there is a bash scripts with it's own data and loops that calls YTT rather than a single YTT call that does everything.

**Comment by mamachanko** [comment]
You could compose it with [yq's ability to split documents into files](https://mikefarah.gitbook.io/yq/usage/split-into-multiple-files#split-single-document-into-files).
